File: 1543a_eb.cpp
/**
 * @file 1543a_eb.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Exciting
 * Bets(https://codeforces.com/problemset/problem/1543/A)
 * @version 0.1
 * @date 2024-08-30
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    short t;
    long long a, b, res;

    cin >> t;
    while (t--) {
        cin >> a >> b;
        if (a == b)
            cout << "0 0\n";
        else {
            res = abs(a - b);
            cout << res << ' ' << min(a % res, res - a % res) << '\n';
        }
    }
    return (EXIT_SUCCESS);
}

File: 1543a_eb.py
"""
@file: 1543a_eb.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Exciting Bets(https://codeforces.com/problemset/problem/1543/A)
@version: 1.0
@date: 2025-06-29

@copyright: Copyright (c) 2025
"""

from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(*args, sep=" ", end="\n"):
    return stdout.write(sep.join(str(x) for x in args) + end)


def main():
    for _ in range(int(input())):
        a, b = [int(x) for x in input().split()]
        gcd = abs(a - b)
        if gcd == 0:
            print(0, 0)
            continue
        print(gcd, min(gcd - a % gcd, a % gcd))


if __name__ == "__main__":
    main()
